pkg/controller/extendeddaemonset/metrics.go

```diff
@@ -0,0 +1,293 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package extendeddaemonset
+
+import (
+	"regexp"
+	"sort"
+
+	datadoghqv1alpha1 "github.com/datadog/extendeddaemonset/pkg/apis/datadoghq/v1alpha1"
+	"github.com/datadog/extendeddaemonset/pkg/controller/metrics"
+	"github.com/datadog/extendeddaemonset/pkg/generated/clientset/versioned"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	ksmetric "k8s.io/kube-state-metrics/pkg/metric"
+)
+
+const (
+	resourceNamePromLabel      = "name"
+	resourceNamespacePromLabel = "namespace"
+
+	extendeddaemonsetCreated                        = "extendeddaemonset_created"
+	extendeddaemonsetStatusDesired                  = "extendeddaemonset_status_desired"
+	extendeddaemonsetStatusCurrent                  = "extendeddaemonset_status_current"
+	extendeddaemonsetStatusReady                    = "extendeddaemonset_status_ready"
+	extendeddaemonsetStatusAvailable                = "extendeddaemonset_status_available"
+	extendeddaemonsetStatusUpToDate                 = "extendeddaemonset_status_uptodate"
+	extendeddaemonsetStatusIgnoredUnresponsiveNodes = "extendeddaemonset_status_ignored_unresponsive_nodes"
```

Comment:
shouldn't we use `eds_` as a prefix instead to avoid getting very long metric names?